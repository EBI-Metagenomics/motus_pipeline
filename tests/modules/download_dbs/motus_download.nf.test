nextflow_workflow {

    name "Test Workflow MOTUS_DOWNLOAD_DB_TEST"
    script "modules/motus.nf"
    workflow "MOTUS_DOWNLOAD_DB"

    test("Should run without failures") {

        when {
            workflow {
                """
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }
    }
}
