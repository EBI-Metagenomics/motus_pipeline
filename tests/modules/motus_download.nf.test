nextflow_workflow {

    name "Test Workflow MOTUS_DB_DOWNLOAD_TEST"
    script "subworkflows/motus_download.nf"
    workflow "MOTUS_DB_DOWNLOAD_TEST"

    test("Should run without failures") {

        when {
            workflow {
                """
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out.motus_db).match()
        }
    }
}
