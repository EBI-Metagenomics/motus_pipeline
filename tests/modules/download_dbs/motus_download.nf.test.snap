{
    "Should run without failures": {
        "content": [
            [
                [
                    "README:md5,4aef627366d259f3812735583ef14516",
                    "db_mOTU_DB_CEN.fasta:md5,48237c3174709b0380cda9e4950993ab",
                    "db_mOTU_DB_CEN.fasta.amb:md5,cabebfd2b6ce61ce442abf2c4272b70b",
                    "db_mOTU_DB_CEN.fasta.ann:md5,026c331cc49329cb2651f6a322cf531b",
                    "db_mOTU_DB_CEN.fasta.annotations:md5,a3705303cf48f7cbe481d5dec3958790",
                    "db_mOTU_DB_CEN.fasta.bwt:md5,0f171a5d2e19ee563ae7550c5eeb15b9",
                    "db_mOTU_DB_CEN.fasta.pac:md5,7bf154d580176e678a93233d61ec9a07",
                    "db_mOTU_DB_CEN.fasta.sa:md5,9f8645aaa8d0751db3eb6f71d1112b92",
                    "db_mOTU_DB_NR.fasta:md5,cdc87352c1e8d609506c9744d3abe0a6",
                    "db_mOTU_DB_NR.fasta.amb:md5,5cc53e6a68fe49321f18cda43881a68a",
                    "db_mOTU_DB_NR.fasta.ann:md5,2b453e321d55c32b8345ab26f5adc584",
                    "db_mOTU_DB_NR.fasta.bwt:md5,c529faec8a4c4c3085057e1037c6b9c8",
                    "db_mOTU_DB_NR.fasta.pac:md5,bd3b1a374dd8a31846057507b7d41521",
                    "db_mOTU_DB_NR.fasta.sa:md5,b1795f309323dbc2a9459b582cf78817",
                    "db_mOTU_MAP_MGCs_to_mOTUs.tsv:md5,2e81eb9a9436cc3114e07858a6b579eb",
                    "db_mOTU_MAP_MGCs_to_mOTUs_in-line.tsv:md5,102946298aa9faca8487983b77deecc5",
                    "db_mOTU_MAP_genes_to_MGCs.tsv:md5,4c03cd58ef08d2aa69b8cae156c67d61",
                    "db_mOTU_bam_header_CEN:md5,f11ebbae7092eaf19732c76f20ebddc8",
                    "db_mOTU_bam_header_NR:md5,dc6c7bcd6c3739068a92465bcbc863b0",
                    "db_mOTU_genes_length_NR:md5,1feb6b2f40d4b5ddfade03c2944fffe7",
                    "db_mOTU_padding_coordinates_CEN.tsv:md5,6a15b99cdc19da2767d7cbbf5bd13851",
                    "db_mOTU_padding_coordinates_NR.tsv:md5,9b7a4318dd61c2b7ba0e4d0aad5a1d8f",
                    "db_mOTU_taxonomy_CAMI.tsv:md5,fee75a872bd6e5804e34d0f8fef4746d",
                    "db_mOTU_taxonomy_meta-mOTUs.tsv:md5,04d3c984d28bacb0d40edfbb5526822a",
                    "db_mOTU_taxonomy_ref-mOTUs.tsv:md5,fc5128fdd70b8b0b97fde8a494493b37",
                    "db_mOTU_taxonomy_ref-mOTUs_short_names.tsv:md5,8cac64c827ed5dab6d739aec1ec9b6f6",
                    [
                        "test1.motus:md5,cc20d6df440b669b06e444f9f1d8977b",
                        "test1_single.fastq:md5,c03f852de2bb09ea386bb1b82e17e21f"
                    ],
                    "db_mOTU_versions:md5,eb11d83b2aadd3ff33d7fb56acb3a5b9",
                    [
                        "mOTUs.profiles.gz:md5,e2cc9d2cb2abae12697ae7df74236d36",
                        "mOTUs.profiles_environments.gz:md5,e47147cb893ef0f9c84e05a5d19e8be8"
                    ]
                ]
            ]
        ],
        "timestamp": "2023-03-21T09:36:21+0000"
    }
}
